polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -1137,13 +1151,17 @@ fn filter_backed_statements_from_disabled_validators<T: shared::Config + schedul
 
 		// Bitmask with the disabled indices within the validator group
 		let disabled_indices = BitVec::<u8, bitvec::order::Lsb0>::from_iter(validator_group.iter().map(|idx| disabled_validators.contains(idx)));
+		let mut validator_indices = bc.validator_indices(core_index_enabled);
 		// The indices of statements from disabled validators in `BackedCandidate`. We have to drop these.
-		let indices_to_drop = disabled_indices.clone() & &bc.validator_indices;
+		let indices_to_drop = disabled_indices.clone() & &validator_indices;
 		// Apply the bitmask to drop the disabled validator from `validator_indices`
-		bc.validator_indices &= !disabled_indices;
+		validator_indices &= !disabled_indices;
+		// Update the backed candidate
+		bc.set_validator_indices(validator_indices);
```

Comment:
This actually siltently removes the core index from the validator_indices bitfield, so the inclusion module won't have access to the core indices in process_candidates()

Reply:
changed this